internal/netconf/interfaces.go

```diff
@@ -2,92 +2,117 @@ package netconf
 
 import (
 	"fmt"
-
-	"github.com/metal-stack/metal-networker/pkg/net"
-
-	"github.com/metal-stack/metal-networker/pkg/exec"
-)
-
-const (
-	// TplFirewallIfaces defines the name of the template to render interfaces configuration for firewalls.
-	TplFirewallIfaces = "interfaces.firewall.tpl"
-	// TplMachineIfaces defines the name of the template to render interfaces configuration for machines.
-	TplMachineIfaces = "lo.network.machine.tpl"
+	"io"
+	"text/template"
 )
 
 type (
-	// CommonIfacesData contains attributes required to render common network interfaces configuration of a bare metal
+	// IfacesData contains attributes required to render network interfaces configuration of a bare metal
 	// server.
-	CommonIfacesData struct {
+	IfacesData struct {
 		Comment  string
 		Loopback Loopback
+		Tenants  []Tenant
 	}
 
-	// MachineIfacesData contains attributes required to render network interfaces configuration of a bare metal
-	// server that functions as 'machine'.
-	MachineIfacesData struct {
-		CommonIfacesData
-	}
-
-	// FirewallIfacesData contains attributes required to render network interfaces configuration of a bare metal
-	// server that functions as 'firewall'.
-	FirewallIfacesData struct {
-		CommonIfacesData
-		Bridge         Bridge
-		EVPNInterfaces []EVPNIface
-	}
-
-	// IfacesValidator defines the base type of an interfaces validator.
-	IfacesValidator struct {
-		path string
-	}
 	// SystemdNetworkdValidator defines the base type of an systemd-networkd validator.
 	SystemdNetworkdValidator struct {
 		path string
 	}
 )
 
-// NewIfacesConfigApplier constructs a new instance of this type.
-func NewIfacesConfigApplier(kind BareMetalType, kb KnowledgeBase, tmpFile string) net.Applier {
-	var data interface{}
-	var validator net.Validator
+type IfacesApplier struct {
+	kind BareMetalType
+	kb   KnowledgeBase
+	d    IfacesData
```

Comment:
IMHO within a function for a narrow scope it is fine to have really short names. For structs I would not apply this and choose "data" over "d".

Reply:
renamed to data